src/shared/components/svg-icon.js

```diff
@@ -6,7 +6,12 @@ import propTypes from 'prop-types';
 /**
  * Object mapping icon names to SVG markup.
  *
- * @typedef {{[name: string]: string}} IconMap
+ * @typedef {Object.<string,string>} IconMap
+ */
+
+/**
+ * @template T
+ * @typedef {import("preact/hooks").Ref<T>} Ref
```

Comment:
Is this sort of thing necessary any time one uses a hook? It seems a little verbose...

Reply:
I don't think so. Type inference should generally avoid the need for this stuff for most of the other hooks. `useRef` is a bit different because of the way it gets initialised. I might be able to find some more character-efficient ways to do this in future.